tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/jdbc/JdbcTableManager.java

```diff
@@ -73,18 +79,77 @@ public TableInstance createImmutable(JdbcTableDefinition tableDefinition, TableH
     {
         TableName tableName = createImmutableTableName(tableHandle);
         LOGGER.debug("creating immutable table {}", tableName);
-        dropTableIgnoreError(tableName); // ultimately we do not want to drop table here - if data did not change
-        createTable(tableDefinition, tableName);
-        JdbcTableDataSource dataSource = tableDefinition.getDataSource();
-        insertData(tableName, dataSource);
+        // TODO: drop and recreate table if the underlying data has changed
+        if (!tableExists(tableName)) {
+            createTable(tableDefinition, tableName);
```

Comment:
Can you maybe extract the following three lines to a separate method?
Maybe `private void createTableWithData(tableDefinition, tableName)`?